### PR TITLE
MOTECH-1760: fixed cycles in test packages

### PR DESCRIPTION
**Pull request by LukSkarDev**
This commit upgrades maven-enforcer-plugin for motech repository to 1.4.1 and  no-package-cycles-enforcer-rule to 1.0.5. From now on cycles will be found also in test packages. All existing test packages cycles were fixed.

### CODE REVIEW

File: modules/tasks/tasks/src/test/java/org/motechproject/tasks/contract/chennel/ChannelRequestTest.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package org.motechproject.tasks.contract;
+package org.motechproject.tasks.contract.chennel;
```
Comment (sebbrudzinski) [inline modules/tasks/tasks/src/test/java/org/motechproject/tasks/contract/chennel/ChannelRequestTest.java]: channel

File: modules/testing-utils/testing-utils/src/main/java/org/motechproject/testing/utils/BasePkgTest.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ public void cleanUp() {
     public abstract String getChrootDirProp();
 
     /**
-     * Reads the script with the given name from the classpath and installs it into the temp directory.
+     * Reads the script with the given name from the classpath and installs it into the jdo directory.
```
Comment (sebbrudzinski) [inline modules/testing-utils/testing-utils/src/main/java/org/motechproject/testing/utils/BasePkgTest.java]: what is jdo directory?

File: modules/testing-utils/testing-utils/src/main/java/org/motechproject/testing/utils/BasePkgTest.java [hunk 3]
```diff
@@ -101,7 +101,7 @@ protected void installScript(String name) throws IOException {
      * @param passPorts whether to pass the test ports as arguments to the script
      * @param attrs the parameters passed to the script, this array will be directly passed to the process builder
      * @return the exit code of the script
-     * @throws IOException if there was an error copying the script into the temp directory
+     * @throws IOException if there was an error copying the script into the jdo directory
```
Comment (sebbrudzinski) [inline modules/testing-utils/testing-utils/src/main/java/org/motechproject/testing/utils/BasePkgTest.java]: ?

File: platform/mds/mds/src/main/java/org/motechproject/mds/annotations/internal/vfs/JndiUrlType.java [hunk 4]
```diff
@@ -34,7 +34,7 @@ public boolean matches(URL url) {
             BundleContext bundleContext = FrameworkUtil.getBundle(JndiUrlType.class).getBundleContext();
             Object jndiService = OSGiServiceUtils.findService(bundleContext, JNDI_SERVICE);
 
-            // copy to the resource to the temp file
+            // copy to the resource to the jdo file
```
Comment (sebbrudzinski) [inline platform/mds/mds/src/main/java/org/motechproject/mds/annotations/internal/vfs/JndiUrlType.java]: jdo file?

File: platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/JarGeneratorServiceImpl.java [hunk 5]
```diff
@@ -159,8 +159,8 @@ private synchronized void regenerateMdsDataBundle(boolean startBundle, String...
         try {
             FileUtils.copyFile(tmpBundleFile, dest);
         } catch (IOException e) {
-            LOGGER.error("Unable to copy the mds-entities bundle to the bundle directory. Installing from temp directory", e);
-            // install from temp directory
+            LOGGER.error("Unable to copy the mds-entities bundle to the bundle directory. Installing from jdo directory", e);
+            // install from jdo directory
```
Comment (sebbrudzinski) [inline platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/JarGeneratorServiceImpl.java]: ??

File: platform/mds/mds/src/test/java/org/motechproject/mds/service/impl/csv/PdfCsvExporterTest.java [hunk 6]
```diff
@@ -70,7 +70,7 @@ public void setUp() {
         CsvTestHelper.mockRecord2Fields(entity);
     }
 
-    // Uncomment the writePdfToFile() calls to write a temp file with the pdf content
+    // Uncomment the writePdfToFile() calls to write a jdo file with the pdf content
```
Comment (sebbrudzinski) [inline platform/mds/mds/src/test/java/org/motechproject/mds/service/impl/csv/PdfCsvExporterTest.java]: ?

File: platform/server/src/main/java/org/motechproject/server/impl/JspBundleLoader.java [hunk 7]
```diff
@@ -180,7 +180,7 @@ private File saveJspToTempFile(InputStream input) throws IOException {
         File tempJspFile = null;
 
         if (tempDir != null && tempDir.isDirectory()) {
-            tempJspFile = new File(tempDir, "temp.jsp");
+            tempJspFile = new File(tempDir, "jdo.jsp");
```
Comment (sebbrudzinski) [inline platform/server/src/main/java/org/motechproject/server/impl/JspBundleLoader.java]: ?

File: modules/testing-utils/testing-utils/src/main/java/org/motechproject/testing/utils/BasePkgTest.java (hunk 2 above)
Comment (LukSkarDev) [inline modules/testing-utils/testing-utils/src/main/java/org/motechproject/testing/utils/BasePkgTest.java]: It had to change when I did some auto refactoring with IntelliJ and I missed it. Will fix it in next commit. The same applies to all strange "temp" to "jdo" changes.

Comment (sebbrudzinski) [inline modules/testing-utils/testing-utils/src/main/java/org/motechproject/testing/utils/BasePkgTest.java]: Okies.

File: platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/JarGeneratorServiceImpl.java [hunk 8]
```diff
@@ -159,7 +159,7 @@ private synchronized void regenerateMdsDataBundle(boolean startBundle, String...
         try {
             FileUtils.copyFile(tmpBundleFile, dest);
         } catch (IOException e) {
-            LOGGER.error("Unable to copy the mds-entities bundle to the bundle directory. Installing from temp directory", e);
+            LOGGER.error("Unable to copy the mds-entities bundle to the bundle directory. Installing from jdo directory", e);
```
Comment (sebbrudzinski) [inline platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/JarGeneratorServiceImpl.java]: this should stay "temp" as well I believe

**Comment by sebbrudzinski** [comment]
Moreover, there are conflicts that must be resolved. Other than that, looks good.

**Pull request merged**